frontend/app.py

```diff
@@ -1604,6 +1607,56 @@ def _create_azure_id_token_response(
     )
 
 
+@create_response.register
+def _create_webdav_token_response(
+    token_request_details: WebDavTokenRequest,
+    canarydrop: Canarydrop,
+    settings: Optional[FrontendSettings] = None,
+) -> WebDavTokenResponse:
+
+    if settings is None:
+        settings = frontend_settings
+
+    if not (
+        settings.WEBDAV_SERVER
+        and settings.CLOUDFLARE_ACCOUNT_ID
+        and settings.CLOUDFLARE_API_TOKEN
+        and settings.CLOUDFLARE_NAMESPACE
+    ):
+        return JSONResponse(
+            {
+                "error_message": "This Canarytokens instance does not have the Network Folder Canarytoken enabled."
+            },
+            status_code=400,
+        )
+    canarydrop.webdav_fs_type = FsType(token_request_details.webdav_fs_type)
+    canarydrop.webdav_server = settings.WEBDAV_SERVER
+    queries.save_canarydrop(canarydrop=canarydrop)
```

Comment:
Do we need to save before generating the password (doesn't look like the password generation can really fail)?